hasher-matcher-actioner/hmalib/common/config.py

```diff
@@ -0,0 +1,282 @@
+    """Delete a config by name only"""
+    assert _TABLE_NAME
+    get_dynamodb().meta.client.delete_item(
+        TableName=_TABLE_NAME,
+        Key={
+            "ConfigType": config_type,
+            "ConfigName": name,
+        },
+    )
+
+
+def delete_config(config: HMAConfig) -> None:
+    """Delete a config"""
+    delete_config_by_name(config.get_config_type(), config.name)
+
+
+def _dynamodb_item_to_config(
+    config_cls: t.Type[TConfig], aws_item: t.Dict[str, t.Any]
+) -> "HMAConfig":
+    """Convert the result of a get_item into a config"""
+    kwargs = {}
+    for field in fields(config_cls):
+        aws_field = aws_item.get(field.name)
+        if aws_field is None:
+            continue  # Hopefully missing b/c default or version difference
+        kwargs[field.name] = _aws_field_to_py(field.type, aws_field)
+    kwargs["name"] = aws_item["ConfigName"]
+    assert aws_item["ConfigType"] == config_cls.get_config_type()
+    return config_cls(**kwargs)
+
+
+def _config_to_dynamodb_item(config) -> t.Dict[str, t.Any]:
+    """
+    Convert a config object into what is what goes into the put_item Item arg
+    """
+    item = {
+        field.name: _py_to_aws_field(field.type, getattr(config, field.name))
+        for field in fields(config)
+    }
+    del item["name"]
+    item["ConfigType"] = config.get_config_type()
+    item["ConfigName"] = config.name
+    return item
+
+
+def _aws_field_to_py(in_type: t.Type[T], aws_field: t.Any) -> T:
+    """
+    Convert an AWS item back into its py equivalent
+
+    This might not even be strictly required, but we check that
+    all the types are roughly what we expect, and convert
+    Decimals back into ints/floats
+    """
+    origin = t.get_origin(in_type)
+    args = t.get_args(in_type)
+
+    check_type = origin
+    if in_type in (int, float):
+        check_type = Decimal
+
+    if not isinstance(aws_field, check_type or in_type):
+        raise HMAConfigSerializationError(
+            "DynamoDB Deserialization error: "
+            f"Expected {in_type} got {type(aws_field)} ({aws_field!r})"
+        )
+
+    if in_type is int:  # N
+        return int(aws_field)  # type: ignore # mypy/issues/10003
+    if in_type is float:  # N
+        return float(aws_field)  # type: ignore # mypy/issues/10003
+    if in_type is Decimal:  # N
+        return aws_field  # type: ignore # mypy/issues/10003
+    if in_type is str:  # S
+        return aws_field  # type: ignore # mypy/issues/10003
+    if in_type is bool:  # BOOL
+        return aws_field  # type: ignore # mypy/issues/10003
+    if in_type is t.Set[str]:  # SS
+        return aws_field  # type: ignore # mypy/issues/10003
+    if in_type is t.Set[int]:  # SN
+        return {int(s) for s in aws_field}  # type: ignore # mypy/issues/10003
+    if in_type is t.Set[float]:  # SN
+        return {float(s) for s in aws_field}  # type: ignore # mypy/issues/10003
+
+    if origin is list:  # L
+        assert len(args) == 1
```

Comment:
Can you explain why multiple args could be passed to the list type?

Reply:
I thought that it would implicitly convert to tuple if you did List[str, str], but it appears to error, so I can safely remove these
```
 t.List[int, str]
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "C:\Python39\lib\typing.py", line 262, in inner
    return func(*args, **kwds)
  File "C:\Python39\lib\typing.py", line 815, in __getitem__
    _check_generic(self, params, self._nparams)
  File "C:\Python39\lib\typing.py", line 199, in _check_generic
    raise TypeError(f"Too {'many' if alen > elen else 'few'} parameters for {cls};"
TypeError: Too many parameters for typing.List; actual 2, expected 1
```